src/main/java/ecommerce/dao/impl/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,184 @@
+            statement.setString(1, user.getName());
+            statement.setString(2, user.getLogin());
+            statement.setString(3, user.getPassword());
+            statement.setLong(5, user.getId());
+            statement.executeUpdate();
+            return user;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update user " + user, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE users SET deleted = TRUE WHERE user_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement
+                         = connection.prepareStatement(query)) {
+            deleteUserFromRoles(id, connection);
+            statement.setLong(1, id);
+            int deleted = statement.executeUpdate();
+            return deleted != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete user with ID " + id, e);
+        }
+    }
+
+    private User getUserFromResultSet(ResultSet resultSet) throws SQLException {
+        Long id = resultSet.getLong("user_id");
+        String name = resultSet.getString("name");
+        String login = resultSet.getString("login");
+        String password = resultSet.getString("password");
+        return new User(id, name, login, password);
+    }
+
+    private void addRolesToUser(User user, Connection connection) throws SQLException {
+        String selectQuery = "SELECT role_id FROM roles WHERE role_name = ?";
+        String addRolesQuery = "INSERT INTO users_roles (user_id, role_id) VALUES (?, ?);";
+        try (PreparedStatement selectStatement
+                     = connection.prepareStatement(selectQuery);
+                 PreparedStatement insertStatement
+                         = connection.prepareStatement(addRolesQuery)) {
+            for (Role role : user.getRoles()) {
+                selectStatement.setString(1, role.getRoleName().name());
+                try (ResultSet resultSet = selectStatement.executeQuery()) {
+                    resultSet.next();
+                    insertStatement.setLong(1, user.getId());
+                    insertStatement.setLong(2, resultSet.getLong("role_id"));
+                    insertStatement.executeUpdate();
+                }
+            }
+        }
+    }
+
+    private Set<Role> getRolesByUserId(Long userId, Connection connection) throws SQLException {
+        String query = "SELECT role_name FROM users_roles "
+                + "JOIN roles USING (role_id) WHERE user_id = ?;";
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, userId);
+            try (ResultSet resultSet = statement.executeQuery()) {
+                Set<Role> roles = new HashSet<>();
+                while (resultSet.next()) {
+                    roles.add(Role.of(resultSet.getString("role_name")));
```

Comment:
what about role_id, where do you set it?

Reply:
I don't need to set role_id to user, role_name is enough. I need role_id only in roles table

Reply:
done